src/provable/from-layout.ts

```diff
@@ -285,6 +294,67 @@ function ProvableFromLayout<
     return (TypeMap as any)[typeData.type].fromFields(fields, aux);
   }
 
+  function emptyValue(typeData: Layout): any {
```

Comment:
this function creates a 'dummy' for an arbitrary type, derived from the layout. this turned out to be necessary, to replace the simpler `emptyValue` function above which just created the value from all-zero field elements. some leaf types have custom `emptyValues`, which aren't all zeroes